src/features/DebugSession.ts

```diff
@@ -386,3 +400,122 @@ export class PickPSHostProcessFeature implements IFeature {
+
+    private command: vscode.Disposable;
+    private languageClient: LanguageClient;
+    private waitingForClientToken: vscode.CancellationTokenSource;
+    private getLanguageClientResolve: (value?: LanguageClient | Thenable<LanguageClient>) => void;
+
+    constructor() {
+        this.command =
+            vscode.commands.registerCommand("PowerShell.PickRunspace", (processId) => {
+                return this.getLanguageClient()
+                           .then((_) => this.pickRunspace(processId), (_) => undefined);
+            }, this);
+    }
+
+    public setLanguageClient(languageClient: LanguageClient) {
+        this.languageClient = languageClient;
+
+        if (this.waitingForClientToken) {
+            this.getLanguageClientResolve(this.languageClient);
+            this.clearWaitingToken();
+        }
+    }
+
+    public dispose() {
+        this.command.dispose();
+    }
+
+    private getLanguageClient(): Thenable<LanguageClient> {
+        if (this.languageClient) {
+            return Promise.resolve(this.languageClient);
+        } else {
+            // If PowerShell isn't finished loading yet, show a loading message
+            // until the LanguageClient is passed on to us
+            this.waitingForClientToken = new vscode.CancellationTokenSource();
+
+            return new Promise<LanguageClient>(
+                (resolve, reject) => {
+                    this.getLanguageClientResolve = resolve;
+
+                    vscode.window
+                        .showQuickPick(
+                            ["Cancel"],
+                            { placeHolder: "Attach to PowerShell host process: Please wait, starting PowerShell..." },
+                            this.waitingForClientToken.token)
+                        .then((response) => {
+                            if (response === "Cancel") {
+                                this.clearWaitingToken();
+                                reject();
+                            }
+                        });
+
+                    // Cancel the loading prompt after 60 seconds
+                    setTimeout(() => {
+                        if (this.waitingForClientToken) {
+                            this.clearWaitingToken();
+                            reject();
+
+                            vscode.window.showErrorMessage(
+                                "Attach to PowerShell host process: PowerShell session took too long to start.");
+                        }
+                    }, 60000);
+                },
+            );
+        }
+    }
+
+    private pickRunspace(processId): Thenable<string> {
+        return this.languageClient.sendRequest(GetRunspaceRequestType, processId).then((response) => {
+            const items: IRunspaceItem[] = [];
+
+            for (const runspace of response) {
+                // Skip default runspace
+                if (runspace.id === 1 || runspace.name === "PSAttachRunspace") {
```

Comment:
I totally just realized that this should only be filtered out IF in the `current` process...

If we're in a different process, we probably want to attach to runspace 1.

Reply:
@adamdriscoll can you fix this up?

Reply:
Yep! Will get another PR in.